### ISSUE TITLE
error location indicated as incorrect source

### ISSUE THREAD

**Issue by coinzdude** [coinzdude/javascript-errors-notifier#12]
Developers can use commonly use console.error for logging errors, but the notifier is catching these currently and then the browser console is incorrectly indicating the source of the error as being from the extension (via code-to-inject.js) instead of the actual source of the error from within the browser debugging console.  See screenshots below showing with and without the extension.

With the extension on, and using "window.console.error = function () { "
the browser incorrectly indicates the error source as below
![image](https://user-images.githubusercontent.com/58857646/226074576-87cef274-f285-42ab-b857-9fc42d8ee9b6.png)

With the extension disabled, the browser console will indicate the proper source.
![image](https://user-images.githubusercontent.com/58857646/226074646-e92e7c17-4d26-4e3f-8bd9-c076d53d26a5.png)

1. Consider a toggle, default off, "Catch console.error logs" which if bypassed would not then override window.console.error which is inducing this issue.

Some other potential tools to use for this problem.

a. console.trace() will log the actual source, but not as an error in the browser console.

---

When an error is thrown as below, when the error uses 'console.error' it causes the error to be incorrectly as being thrown from code-to-inject.js

Need to see if we can detect the error due to console.error and specially handle, to avoid the browser thinking the error is from code-to-inject.js

if the js uses 'console.error' the above issue is triggered.

a more robust error is here
```

let mydata = {} as any;
            setTimeout(function () {
                console.error("There was an error!!!!." + mydata.Error);
            }, 3000);
```

**Comment by coinzdude** [comment]
Alternatively, consider if we can get the stack trace in the above and print the stack to appropriately indicate the error source file

**Comment by coinzdude** [comment]
Consider using a 'getStack()' function here to induce an error and get the stack to log and indicate the real stack of the source

**Comment by coinzdude** [comment]
fixed with d998da4dba4fe74f560bc89edcec40ce3e28d28d new option to ignore console.error
